src/server/scripts/Shadowlands/SepulcherOfTheFirstOnes/boss_anduin_wrynn.cpp

```diff
@@ -0,0 +1,4307 @@
+    { -3839.6390f, -2750.8890f, 91.53032f, 1.1935f }, // Fourth Right
+    { -3811.0625f, -2679.6528f, 91.53031f, 4.3362f }, // First left
+    { -3789.8125f, -2700.5088f, 91.53032f, 3.5383f }, // Second Left
+    { -3789.2812f, -2729.3160f, 91.53032f, 2.7669f }, // Third Left
+    { -3810.4290f, -2751.0903f, 91.53032f, 1.9584f } // Fourth Left
+};
+
+Position const BeaconOfHopeSpawnPosition = { -3825.0417f, -2715.3923f, 91.3568f, 0.0f };
+
+Position const ChestLootSpawnPosition = { -3840.9915f, -2741.7847f, 91.26521f, 1.334929f };
+
+template<typename T>
+struct Timers : public std::list<T>
+{
+    Timers() = default;
+
+    Timers(const std::initializer_list<T>& _list) :
+        std::list<T>(_list) {};
+
+    T nextTimer()
+    {
+        if (this->empty())
+            throw std::logic_error("Timers is empty!");
+
+        T timer = this->front();
+        if (this->size() > 1)
+            this->pop_front();
+
+        return timer;
+    }
+};
+
+Timers<uint32> HopeBreakerP1{ 31900, 28000, 29900, 29900 };
+Timers<uint32> HopeBreakerP2{ 22000, 33300, 29000, 29000 };
+Timers<uint32> HopeBreakerP2Mythic{ 25000, 33000, 29000, 29100 };
+Timers<uint32> BefouledBarrierTimerP1{ 51900, 48000 };
+Timers<uint32> BefouledBarrierTimerP2{ 47000 };
+Timers<uint32> BlasphemyAll{ 49900, 54900 };
+Timers<uint32> DominationWordPainP1{ 13000, 13000, 10000, 15000, 13100, 12900, 13000, 13900, 12200, 14800 };
+Timers<uint32> DominationWordPainP2{ 13000, 13000, 17700, 8100, 13000, 13000, 14400, 11200, 12200 };
+Timers<uint32> WickedStarP1{ 35000, 30000 };
+Timers<uint32> WickedStarP2{ 39000, 26000, 30500, 19000 };
+Timers<uint32> WickedStarP2Mythic{ 39000, 26000, 30900, 19100 };
+
+class ActivateGhouls : public BasicEvent
+{
+public:
+    ActivateGhouls(Creature* summoner, Creature* owner)
+        : _summonerGuid(summoner->GetGUID()), _owner(owner) { }
+
+    bool Execute(uint64 /*time*/, uint32 /*diff*/) override
+    {
+        if (Creature* _summoner = ObjectAccessor::GetCreature(*_owner, _summonerGuid))
+        {
+            if (Unit* target = _summoner->AI()->SelectTarget(SelectTargetMethod::Random, 0))
+            {
+                _owner->SetReactState(REACT_AGGRESSIVE);
+                _owner->AI()->AttackStart(target);
+            }
+        }
+        return true;
+    }
+
+private:
+    ObjectGuid _summonerGuid;
+    Creature* _owner;
+};
+
+// Anduin Wrynn - 181954
+struct boss_anduin_wrynn : public BossAI
+{
+    boss_anduin_wrynn(Creature* creature) : BossAI(creature, DATA_ANDUIN_WRYNN),
+        _slayTextOnCooldown(false), _dominationWordCount(0), _intermissionsDone(0), _encounterEnded(false) { }
+
+    void HandleIntroduction()
+    {
+        if (instance->GetData(DATA_ANDUIN_WRYNN_INTRODUCTION) == DONE)
+        {
+            me->RemoveUnitFlag(UNIT_FLAG_NOT_ATTACKABLE_1);
+            me->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+            me->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_NPC);
```

Comment:
me->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
me->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_NPC);

these are not required, SetImmuneToAll handles it all.